assign_rights/tests.py

```diff
@@ -119,6 +119,21 @@ def test_calculate_dates(self):
         for granted in shell.rightsgranted_set.all():
             self.check_object_dates(granted, request_start_date, request_end_date)
 
+    def test_create_json(self):
+        shell = random.choice(RightsShell.objects.all())
+        request_end_date = random_date().isoformat()
+        request_start_date = random_date().isoformat()
+        start_date, end_date = self.assembler.calculate_dates(shell, request_start_date, request_end_date)
+        serialized = self.assembler.create_json(shell, start_date, end_date)
+
+        granted_list = []
+        for granted in shell.rightsgranted_set.all():
+            start_date, end_date = self.assembler.calculate_dates(granted, request_start_date, request_end_date)
+            granted_list.append(self.assembler.create_json(granted, start_date, end_date))
+        for item in granted_list:
+            serialized["rights_granted"].append(item)
```

Comment:
I think you could potentially wrap the logic in lines 123-127 in a `for` loop with works against a list of the classes you want to test (RightsShell and RightsGranted) and just dispense with this code. After all, you're really testing whether the serialization works as expected, not the whole run method.